pkg/etcd/etcd_test.go

```diff
@@ -0,0 +1,176 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/zapr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	commonLabels "github.com/medik8s/common/pkg/labels"
+)
+
+var guardLabel = map[string]string{"app": "guard"}
+
+var _ = Describe("Check etcd disruptions", func() {
+	var (
+		cl client.WithWatch
+	)
+	zapLogger, _ := zap.NewProduction()
+	log := zapr.NewLogger(zapLogger)
```

Comment:
we need to write to GinkWriter, and better time formatting would be great as well :) See e.g. https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/suite_test.go#L104-L108